pkg/controller/control.go

```diff
@@ -208,14 +204,14 @@ func (c *Controller) Expand(size int64) error {
 			}
 		}
 
-		if size < c.size {
-			logrus.Errorf("Cannot expand to a smaller size %v", size)
-			return
-		} else if size == c.size {
-			logrus.Infof("Controller %v is already expanded to size %v", c.Name, size)
-			return
-		}
+		expanded := false
```

Comment:
Move this to the beginning of the function. Otherwise, if the code before this function exits, the lock will never be released (this won't happen in this case but it's better to follow the pattern).